lib/calculator/parser.rb

```diff
@@ -63,7 +63,61 @@ def shift_tokens
 
     # Entry point for the parser, parses an <em>expression</em>
     def parse_expression
+
       # TODO implement me
+      def parse_expression
+
+        ExpressionNode.new(parse_term, parse_expression_prime)
+      end
+
+      def parse_expression_prime
+            case @tokens.first
+              when AddOpToken
+                ExpressionPrimeNode.new(shift_tokens, parse_expression_prime, parse_term)
+              when SubtractOpToken
+                ExpressionPrimeNode.new(shift_tokens, parse_expression_prime, parse_term)
+
+              else
+                ExpressionPrimeNode.new
+
+            end
+        def parse_term
+            TermNode.new(parse_term_prime, parse_factor)
+
+
+
+        end
+
+        def parse_factor
+            FactorNode.new(parse_term)
+
```

Comment:
According to the grammar, the rule for factor is 'factor -> base { ExponentOpToken factor }', so this doesn't match that rule.